File: packages/sdk-android/src/main/java/app/eodin/deeplink/DeferredParamsResult.kt
package app.eodin.deeplink

/**
 * Result container for deferred deep link parameters.
 *
 * @property path The deep link path to navigate to (e.g., "product/12345")
 * @property resourceId The resource ID from the original link
 * @property metadata Additional metadata if any
 */
data class DeferredParamsResult(
    val path: String?,
    val resourceId: String?,
    val metadata: Map<String, Any>?
) {
    /**
     * Whether deferred parameters were found.
     */
    val hasParams: Boolean
        get() = path != null
}

File: packages/sdk-flutter/lib/src/models/deferred_params_result.dart
/// Result of deferred parameters check
class DeferredParamsResult {
  /// The deep link path to navigate to
  final String? path;

  /// The resource ID from the URL
  final String? resourceId;

  /// Additional metadata
  final Map<String, dynamic>? metadata;

  /// Creates a new [DeferredParamsResult]
  const DeferredParamsResult({
    this.path,
    this.resourceId,
    this.metadata,
  });

  /// Whether this result contains deferred parameters
  bool get hasParams => path != null || resourceId != null;

  /// Creates a [DeferredParamsResult] from JSON
  factory DeferredParamsResult.fromJson(Map<String, dynamic> json) {
    return DeferredParamsResult(
      path: json['path'] as String?,
      resourceId: json['resourceId'] as String?,
      metadata: json['metadata'] as Map<String, dynamic>?,
    );
  }

  /// Converts this result to JSON
  Map<String, dynamic> toJson() {
    return {
      'path': path,
      'resourceId': resourceId,
      'metadata': metadata,
    };
  }

  @override
  String toString() {
    return 'DeferredParamsResult(path: $path, resourceId: $resourceId, metadata: $metadata)';
  }
}
